GPU/Workflow/src/GPUWorkflowSpec.cxx

```diff
@@ -410,51 +411,19 @@ DataProcessorSpec getGPURecoWorkflowSpec(gpuworkflow::CompletionPolicyData* poli
       }
       if (specconfig.zsDecoder) {
         std::vector<InputSpec> filter = {{"check", ConcreteDataTypeMatcher{gDataOriginTPC, "RAWDATA"}, Lifetime::Timeframe}};
-        for (auto const& ref : InputRecordWalker(pc.inputs(), filter)) {
-          const DataHeader* dh = DataRefUtils::getHeader<DataHeader*>(ref);
-          if (dh->payloadSize == 0 && dh->subSpecification == 0xDEADBEEF) {
-            LOG(INFO) << "Received 0xDEADBEEF message with no RAW input, performing dummy processing on emtpry input data";
-            continue; // Dummy message inserted if there is no input, just ignore
-          }
-          const gsl::span<const char> raw = pc.inputs().get<gsl::span<char>>(ref);
-          o2::framework::RawParser parser(raw.data(), raw.size());
-
-          const unsigned char* ptr = nullptr;
-          int count = 0;
-          rdh_utils::FEEIDType lastFEE = -1;
-          int rawcru = 0;
-          int rawendpoint = 0;
-          size_t totalSize = 0;
-          for (auto it = parser.begin(); it != parser.end(); it++) {
-            const unsigned char* current = it.raw();
-            const RAWDataHeader* rdh = (const RAWDataHeader*)current;
-            if (current == nullptr || it.size() == 0 || (current - ptr) % TPCZSHDR::TPC_ZS_PAGE_SIZE || o2::raw::RDHUtils::getFEEID(*rdh) != lastFEE) {
-              if (count) {
-                tpcZSmetaPointers[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(ptr);
-                tpcZSmetaSizes[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(count);
-              }
-              count = 0;
-              lastFEE = o2::raw::RDHUtils::getFEEID(*rdh);
-              rawcru = o2::raw::RDHUtils::getCRUID(*rdh);
-              rawendpoint = o2::raw::RDHUtils::getEndPointID(*rdh);
-              //lastFEE = int(rdh->feeId);
-              //rawcru = int(rdh->cruID);
-              //rawendpoint = int(rdh->endPointID);
-              if (it.size() == 0 && tpcZSmetaPointers[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].size()) {
-                ptr = nullptr;
-                continue;
-              }
-              ptr = current;
-            } else if (ptr == nullptr) {
-              ptr = current;
-            }
-            count++;
-          }
-          if (count) {
-            tpcZSmetaPointers[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(ptr);
-            tpcZSmetaSizes[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(count);
-          }
-        }
+        auto isSameRdh = [](const char* left, const char* right) -> bool {
+          return o2::raw::RDHUtils::getCRUID(left) == o2::raw::RDHUtils::getCRUID(right) && o2::raw::RDHUtils::getEndPointID(left) == o2::raw::RDHUtils::getEndPointID(right) && o2::raw::RDHUtils::getFEEID(left) == o2::raw::RDHUtils::getFEEID(right);
```

Comment:
I'm sorry, there is no line width limit?

Reply:
I think its fine, and would stick to what clang format detects to be fixed. But maybe its good enough to check only the FeeId, what do you think @davidrohr ?

Reply:
Have changed this to only check the Fee Id, this also makes the line shorter